### PR TITLE
section refactoring

### PR DESCRIPTION
**Pull request by veggiemonk**
## To Do

- [ ] Name sections
- [ ] Refactor contents
- [ ] Arrange sections accordingly

Optional steps in this PR?
- [ ] (Re)move listed items accordingly
---

Ok guys, all the comments/ideas are gathered here from all the discussions we had. If you see something is missing please let us know. 

### The `Tools` section was restructured according to #397 and #393 

Also, i want to remove this Twitter section, it doesn't bring any value anymore. It was useful back when Docker was totally unknow for most IT people but not anymore. 

By the way, I think the thing is a bit too serious for my taste.
So I wanted  to try something and instead  of icons I used emojis. Feel free to change them and try others if you like. After all this list is everybody's project and it's better if we have fun while doing it. And we don't use enough emojis!!! 😄 
here are a few that might be of interest: 🎥 for related videos,  📖 for related articles, 💣 for beta.

Feel free to check out this branch and add commits to it. 
``` bash
# in your local repo
git checkout refactor/section

# (make changes)
git add .
git commit -m "your commit message here"
git push
```

# If all of us agree on the name of the section, **before merging it**, we should at least order the section according to the menu. What do you think?

### CODE REVIEW

**Comment by Moshe-Immerman** [comment]
LGTM 

- If we are removing the twitter section, we should probably remove the list of blogs too as most aren't updated that often, or very docker specific - I think the links to specific posts on those blogs are more useful.
- $$$ --> 💲 
- Another option for beta: 🚧
- And for legacy: 💀

File: README.md [hunk 1]
```diff
@@ -21,51 +21,69 @@ All the links are monitored and tested with [awesome_bot](https://github.com/dkh
...
+		- [Registry](#hosting-images-registries)
+	- Development with Docker
+		- [API Client](#developer-tools)
+		- [CI/CD](#continuous-integration--continuous-delivery)
+		- [Development Environment](#development-environments)
+		- [Garbage Collection](#garbage-collection)
+		- [Serverless](#serverless)
+		- [Testing](#testing)
+		- [Wrappers](#local-container-manager)
+		- [Utilities](#utilities) <- must be split
```
Comment (veggiemonk) [inline README.md]: To be split accross other sections

Comment (vegasbrianc) [inline README.md]: @veggiemonk I agree with removing Twitter and re-org of the section, LGTM!

File: README.md [hunk 2]
```diff
@@ -21,51 +21,69 @@ All the links are monitored and tested with [awesome_bot](https://github.com/dkh
...
-	- [Brazilian](#brazilian)
-	- [Chinese](#chinese)
-	- [English](#english)
-	- [Russian](#russian)
-
+	- [Videos](#videos)
+		- [Main Account](#main-account)
+		- [Useful videos](#useful-videos)
+	- ~~[Interesting Twitter Accounts](#interesting-twitter-accounts)~~
+
```
Comment (devhkr) [inline README.md]: I also agree on removing Twitter links, but at least some blogs should remain e.g. The New Stack. In addition, blog sites have links to their social media accounts.

File: README.md [hunk 3]
```diff
@@ -21,51 +21,69 @@ All the links are monitored and tested with [awesome_bot](https://github.com/dkh
...
-- [Communities and Meetups](#communities-and-meetups)
-	- [Brazilian](#brazilian)
-	- [Chinese](#chinese)
-	- [English](#english)
-	- [Russian](#russian)
-
+	- [Videos](#videos)
+		- [Main Account](#main-account)
+		- [Useful videos](#useful-videos)
+	- ~~[Interesting Twitter Accounts](#interesting-twitter-accounts)~~
```
Comment (devhkr) [inline README.md]: I also agree on removing Twitter links, but at least some blogs should remain e.g. The New Stack. In addition, blog sites have links to their social media accounts.

File: README.md [hunk 4]
```diff
@@ -21,51 +21,69 @@ All the links are monitored and tested with [awesome_bot](https://github.com/dkh
...
 - [Useful Resources](#useful-resources)
+	- Awesome Lists
+	- [Communities and Meetups](#communities-and-meetups)
+		- [Brazilian](#brazilian)
+		- [Chinese](#chinese)
+		- [English](#english)
+		- [Russian](#russian)
 	- [Good Tips](#good-tips)
+	- [Raspberry Pi / ARM](#raspberry-pi--arm)
 	- [Newsletter](#newsletter)
```
Comment (devhkr) [inline README.md]: I think we should remove the newsletter subsection as the purpose of subscribing one is to aggregate blog posts. Furthermore, blogs offer RSS feeds.

File: README.md [hunk 5]
```diff
@@ -21,51 +21,69 @@ All the links are monitored and tested with [awesome_bot](https://github.com/dkh
...
-	- [Useful videos](#useful-videos)
-- [Interesting Twitter Accounts](#interesting-twitter-accounts)
-- [Communities and Meetups](#communities-and-meetups)
-	- [Brazilian](#brazilian)
-	- [Chinese](#chinese)
-	- [English](#english)
-	- [Russian](#russian)
-
+	- [Videos](#videos)
+		- [Main Account](#main-account)
```
Comment (devhkr) [inline README.md]: As blog sites have links to their other social media accounts, this subsection can also be removed.

Comment (devhkr) [inline README.md]: As blog sites have links to their other social media accounts, this subsection can also be removed.

File: README.md (hunk 3 above)
Comment (devhkr) [inline README.md]: I also agree on removing Twitter links, but at least some blogs should remain e.g. The New Stack. In addition, blog sites have links to their social media accounts.

File: README.md (hunk 4 above)
Comment (devhkr) [inline README.md]: I think we should remove the newsletter subsection as the purpose of subscribing one is to aggregate blog posts. Furthermore, blogs offer RSS feeds.

**Comment by devhkr** [comment]
I like the emojis. 
By the way, if we have a section called 'Services based on Docker ($$$)', why do we need a :money_mouth_face: emoji? So, it can look like 'Services based on Docker :money_mouth_face:'?

File: README.md [hunk 6]
```diff
@@ -21,51 +21,69 @@ All the links are monitored and tested with [awesome_bot](https://github.com/dkh
...
+		- [Useful videos](#useful-videos)
+	- ~~[Interesting Twitter Accounts](#interesting-twitter-accounts)~~
+
+# Legend
+- Commercial 🤑
+- Beta 🤕
+- Terminal 🤓
+- Web Based 😎
```
Comment (devhkr) [inline README.md]: I like the emojis.
But if we have a section called 'Services based on Docker ($$$)', why do we need a :money_mouth_face: emoji then?

Comment (veggiemonk) [inline README.md]: Good points. I think there is a difference between service and close/open sources. But depending on the use cases we should move forward with this. Let's have a look a do what make sense.

File: README.md (hunk 1 above)
Comment (agebhar1) [inline README.md]: One question about the order of each subsections should we stay by lexicographic order?

File: README.md (hunk 3 above)
Comment (agebhar1) [inline README.md]: Maybe 'Essential ...'/'Leading ...' instead of 'Interesting ...'?

File: README.md (hunk 6 above)
Comment (agebhar1) [inline README.md]: What's your opinion about moving emoji/labels out of pure refactoring and start a new improvement issue about UX?

**Comment by Moshe-Immerman** [comment]
My plan with #406  was to go top down first reordering existing sections, and then populating those sections (e.g Composition and Base Tools) and then moving to the next high level section and finally moving some items around so they fit better into the new structure. 

I have pushed https://github.com/Moshe-Immerman/awesome-docker/commit/ba09738ad3911845890977b7a1adc02ddc994974 which is a pure reorder of the container operations section. 

My next commit https://github.com/veggiemonk/awesome-docker/pull/406/commits/0d797eeb516bc9bbb0d0684fee7c6b807d12ce63 was populating the new composition section from existing sections - Should I push it?

File: README.md (hunk 1 above)
Comment (veggiemonk) [reply README.md]: By default yes, unless there is a reason to order them differently. I cannot think of one right now

File: README.md [hunk 7]
```diff
@@ -119,6 +134,117 @@ Docker EE is on the same code base as Docker CE, so also built from Moby, with c
...
+* [Docket](https://github.com/netvarun/docket) - Custom docker registry that allows for lightning fast deploys through bittorrent by [@netvarun](https://github.com/netvarun/)
+* [dockit](https://github.com/humblec/dockit) - Do docker actions and Deploy gluster containers! By [@humblec](https://github.com/humblec)
+* [Longshoreman](https://github.com/longshoreman/longshoreman) - Longshoreman automates application deployment using Docker. Just create a Docker repository (or use a service), configure the cluster using AWS or Digital Ocean (or whatever you like) and deploy applications using a Heroku-like CLI tool. By [longshoreman](https://github.com/longshoreman)
+* [rocker-compose](https://github.com/grammarly/rocker-compose) - Docker composition tool with idempotency features for deploying apps composed of multiple containers. By [@grammarly](grammarly)
+* [Zodiac](https://github.com/CenturyLinkLabs/zodiac) - A lightweight tool for easy deployment and rollback of dockerized applications. By [@CenturyLinkLabs][CenturyLinkLabs]
+
+## Monitoring
+## Networking
```
Comment (veggiemonk) [inline README.md]: WTF is that?

**Comment by veggiemonk** [comment]
@Moshe-Immerman yes push your commits. I think it would be easier to start working from your work than to try to figure out what changes... Personally I can't.

**Comment by veggiemonk** [comment]
Hi guys, I think that @Moshe-Immerman did a lot of work on the PR #406 
Personally, the changes are too many for me to keep track or verify. So I propose that we work on that PR and forget about this one. The name of the section seems like something we all agree on. It would be faster to be done with this refactor that is taking waaaaay too long. Projects are pilling up waiting to be on the list and that's not good for this project. 

Here is my proposal:

Let's bring the PR #406 to something we can merge to `master` and improve from there. Just check on the github pages that links are working properly. if we missed one or two project during the refactor they can still be found in the git history so no worries. 
@vegasbrianc @agebhar1 @devhkr

**Pull request merged**